src/plugins/intel_cpu/src/nodes/transpose.cpp

```diff
@@ -164,29 +140,15 @@ void Transpose::prepareParams() {
                                             memory::format_tag::acdb);
         src_blocked->Create(DnnlExtensionUtils::makeDescriptor(newDesc), srcMemPtr->GetData(), false);
 
-        impl_desc_type impl_type = getSelectedPrimitiveDescriptor()->getImplementationType();
-        TransposeAsReorderKey key = {src_blocked->GetPrimitive().get_desc(), dst_blocked->GetPrimitive().get_desc()};
-        auto builder = [&engine, &impl_type](const TransposeAsReorderKey& key) -> std::shared_ptr<dnnl::primitive> {
-            dnnl::primitive_attr attr;
-            reorder::primitive_desc pd = dnnl::reorder::primitive_desc(engine, key.src, engine, key.dest, attr, true);
-
-            if (!pd)
-                return nullptr;
-            auto info = pd.impl_info_str();
-            impl_type = parse_impl_name(info);
-            return std::make_shared<dnnl::reorder>(pd);
-        };
-
-        auto cache = getRuntimeCache();
-        auto result = cache->getOrCreate(key, builder);
-
-        if (!result.first) {
+        auto result = getReorderPrim(getRuntimeCache(), getEngine(), src_blocked->GetPrimitive().get_desc(), dst_blocked->GetPrimitive().get_desc());
+        if (!result) {
```

Comment:
Making `src_blocked` and `dst_blocked` memory objects is not necessary at all. Memory descriptors could be retrieved from the memory objects on edges directly. Actually this is the same thing that is fixed in PR https://github.com/openvinotoolkit/openvino/pull/14366. Could you please refactor this part as well?

Reply:
OK, done!

Reply:
@maxnick Please review again, Thx!